### PR TITLE
[oneDPL][ranges][sycl] A fix for header inclusion order in case of ranges

### PR DESCRIPTION
**Pull request by MikeDvorskiy**
A fix for header inclusion order (`<ranges> / <execution>`)
Also, a couple of tests to cover that situation was added.

### CODE REVIEW

File: include/oneapi/dpl/pstl/hetero/utils_hetero.h [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+namespace oneapi
+{
+namespace dpl
+{
+namespace __internal
+{
+
+template <typename _Pred>
+struct equal_predicate
```
Comment (timmiesmith) [inline include/oneapi/dpl/pstl/hetero/utils_hetero.h:32]: Do the `equal_predicate` and `adjacent_find_fn` names need to be uglified?  I.e., `equal_predicate` replaced with `__equal_predicate`?

File: include/oneapi/dpl/pstl/hetero/utils_hetero.h [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            __chosen_for_max = __a;
+        auto __result = _ReduceValueType{get<0>(__chosen_for_min), get<1>(__chosen_for_max), get<2>(__chosen_for_min),
+                                         get<3>(__chosen_for_max)};
+
+        return __result;
+    }
+};
+
+template <typename _Predicate>
+struct acc_handler_count
```
Comment (timmiesmith) [inline include/oneapi/dpl/pstl/hetero/utils_hetero.h:135]: Another candidate for uglification if needed.

File: include/oneapi/dpl/pstl/hetero/utils_hetero.h (hunk 1 above)
Comment (MikeDvorskiy) [reply include/oneapi/dpl/pstl/hetero/utils_hetero.h:32]: That stuff was been just moved to another file.
Probably, to fix qualification issue everywhere does it make sense to prepare a separate patch?

File: include/oneapi/dpl/pstl/hetero/utils_hetero.h (hunk 2 above)
Comment (MikeDvorskiy) [reply include/oneapi/dpl/pstl/hetero/utils_hetero.h:135]: see my comment above

File: include/oneapi/dpl/pstl/hetero/utils_hetero.h (hunk 1 above)
Comment (timmiesmith) [reply include/oneapi/dpl/pstl/hetero/utils_hetero.h:32]: Ok. I see the namespace of these structs wasn't even changed by the move.  I agree that fixing the uglification issue everywhere would be best done as a separate patch then.

**Review by dsobolev-dev — COMMENTED** [review]
Seems done: https://github.com/oneapi-src/oneDPL/blob/d91bac87382bf59b66ca7e08706d7bf7322eb4c6/include/oneapi/dpl/ranges#L20

**Review by dsobolev-dev — APPROVED** [review]
Looks good to me

**Pull request merged**